src/core/or/connection_or.c

```diff
@@ -437,7 +437,15 @@ connection_or_state_publish(const or_connection_t *conn, uint8_t state)
 
   msg.type = ORCONN_MSGTYPE_STATE;
   msg.u.state.gid = conn->base_.global_identifier;
-  msg.u.state.proxy_type = conn->proxy_type;
+  if (conn->is_pt) {
+    /* Do extra decoding because conn->proxy_type indicates the proxy
+     * protocol that tor uses to talk with the transport plugin,
+     * instead of PROXY_PLUGGABLE. */
+    tor_assert(conn->proxy_type != PROXY_NONE);
```

Comment:
Can we make this tor_assert_nonfatal() ? I'd hate to introduce a new crash bug at this stage in the development process.

Reply:
Sure.